### ISSUE TITLE
Score ranges

### ISSUE THREAD

**Issue by kochkinaelena** [castorini/pygaggle#94]
Hi,

This is a question rather than an issue. I was wondering what is the expected range of Pygaggle score? 

From the paper ([Document Ranking with a Pretrained Sequence-to-Sequence Model](https://arxiv.org/pdf/2003.06713.pdf)) it comes across as if it should be the probability of a relevant label obtained from softmax layer, so [0,1]. However, when I am using it on my dataset (not from the paper) with Pyserini as described in your [main readme](https://github.com/castorini/pygaggle) and getting scores from reranked[i].score in the range [-1.47, -0.04].  Could you explain why this is happening? Is it as intended?  

Also, can I use these pygaggle scores to identify relevance threshold and filter out pairs with low scores as irrelevant across different datasets?  

Many thanks,
Elena

**Comment by lintool** [comment]
@ronakice @rodrigonogueira4 nudge?

**Comment by ronakice** [comment]
Hey @kochkinaelena! 

1) The scores we return are log probabilities which is why you'd see that they are less than 0. You can find the specific line here:
https://github.com/castorini/pygaggle/blob/73539eab8a989675472ec3e0d44b5bf3a2e29216/pygaggle/rerank/transformer.py#L57
We chose to go with it because it is what we went with in our initial T5 implementation.
2) Converting to probabilities will probably be useful here, since they are easier to interpret and threshold with. Then, you can finetune to find the best "threshold" in some random subset (We've seen it work with a few datasets).

**Comment by kochkinaelena** [comment]
Hi @ronakice,

Thank you for your response, this clarifies things!

Best regards,
Elena
